script.speedtester/default.py

```diff
@@ -0,0 +1,939 @@
+        self.addControl(self.textbox)
+        self.displayButtonRun()
+        self.displayButtonClose()
+        self.setFocus(self.button_run)
+
+    def onAction(self, action):
+        if action == 10 or action == 92:
+            self.saveClose()
+
+    def displayButtonRun(self, function="true"):
+        if (function == "true"):
+            button_run_glowx = int((self.screenx / 3) - (300 / 2))
+            button_run_glowy = int((self.screeny / 3) - (122 / 2) + 50)
+
+            self.button_run_glow = xbmcgui.ControlImage(button_run_glowx, button_run_glowy, 300, 122, '', aspectRatio=0)
+            self.addControl(self.button_run_glow)
+            self.button_run_glow.setVisible(False)
+            self.button_run_glow.setImage(self.image_button_run_glow)
+            self.button_run_glow.setAnimations([
+                ('conditional', 'effect=fade start=0 time=1000 condition=true pulse=true')
+            ])
+
+            self.button_run = xbmcgui.ControlButton(button_run_glowx, button_run_glowy, 300, 122, "[B]Run Speedtest[/B]",focusTexture=self.image_button_run,noFocusTexture=self.image_button_run, alignment=2 | 4,textColor='0xFF000000', focusedColor='0xFF000000',shadowColor='0xFFCCCCCC', disabledColor='0xFF000000')
+
+            self.addControl(self.button_run)
+            self.setFocus(self.button_run)
+            self.button_run.setVisible(False)
+            self.button_run.setAnimations([
+                ('conditional',
+                'effect=fade start=100 end=0 time=300 condition=!Control.IsEnabled(%d)' % self.button_run.getId())
+            ])
+            self.button_run_ID = self.button_run.getId()
+            self.button_run.setEnabled(True)
+            self.button_run.setVisible(True)
+            self.button_run_glow.setEnabled(True)
+            self.button_run_glow.setVisible(True)
+        else:
+            self.button_run.setEnabled(False)
+            self.button_run.setVisible(False)
+            self.button_run_glow.setEnabled(False)
+            self.button_run_glow.setVisible(False)
+
+
+    def displayButtonClose(self, function="true"):
+        if (function == "true"):
+            self.button_close_glow = xbmcgui.ControlImage(880, 418, 300, 122, '', aspectRatio=0)
+            self.addControl(self.button_close_glow)
+            self.button_close_glow.setVisible(False)
+            self.button_close_glow.setImage(self.image_button_run_glow)
+            self.button_close_glow.setAnimations([
+                ('conditional',
+                'effect=fade start=0 time=1000 delay=2000 pulse=true condition=Control.IsVisible(%d)' % self.button_close_glow.getId())
+            ])
+
+            self.button_close = xbmcgui.ControlButton(99999, 99999, 300, 122, "[B]Close[/B]",focusTexture=self.image_button_run,noFocusTexture=self.image_button_run, alignment=2 | 4,textColor='0xFF000000', focusedColor='0xFF000000',shadowColor='0xFFCCCCCC', disabledColor='0xFF000000')
```

Comment:
Hardcoded close

Reply:
Not sure what you mean here. Can you elaborate?

Reply:
"Close" has to be localized